presto-tests/src/test/java/com/facebook/presto/server/TestServerInfoResource.java

```diff
@@ -43,22 +49,43 @@
 {
     private HttpClient client;
     private DistributedQueryRunner queryRunner;
+    private ThriftCodecManager thriftCodeManager;
+
+    private NodeState getNodeState(URI uri, boolean useThriftEncoding, Protocol thriftProtocol)
+    {
+        Request.Builder requestBuilder = useThriftEncoding ? ThriftRequestUtils.prepareThriftGet(thriftProtocol) : getJsonTransportBuilder(prepareGet());
+        Request request = requestBuilder.setHeader(PRESTO_USER, "user").setUri(uri).build();
+        if (useThriftEncoding) {
+            return client.execute(request, new ThriftResponseHandler<>(thriftCodeManager.getCodec(NodeState.class))).getValue();
+        }
+        else {
+            requestBuilder = getJsonTransportBuilder(prepareGet());
+            return client.execute(request, createJsonResponseHandler(jsonCodec(NodeState.class)));
+        }
+    }
+
+    @DataProvider
+    public Object[][] thriftEncodingToggle()
+    {
+        return new Object[][] {{true, Protocol.BINARY}, {true, Protocol.COMPACT}, {true, Protocol.FB_COMPACT}, {false, null}};
+    }
 
     @BeforeClass
     public void setup()
             throws Exception
     {
         this.client = new JettyHttpClient();
+        this.thriftCodeManager = new ThriftCodecManager();
     }
 
     @AfterClass(alwaysRun = true)
     public void teardown()
     {
         closeQuietly(client);
-        client = null;
+        this.client = null;
     }
 
-    @AfterMethod
+    @AfterGroups
```

Comment:
curious about this change. I think this annotation makes sense when you are specifying the group as well in `@AfterGroups`